release-notes/3.0/preview/api-diff/preview7/3.0-preview7_System.Threading.Channels.md

```diff
@@ -0,0 +1,66 @@
++        Wait = 0,
++    }
++    public sealed class BoundedChannelOptions : ChannelOptions {
++        public BoundedChannelOptions(int capacity);
++        public int Capacity { get; set; }
++        public BoundedChannelFullMode FullMode { get; set; }
++    }
++    public static class Channel {
++        public static Channel<T> CreateBounded<T>(int capacity);
++        public static Channel<T> CreateBounded<T>(BoundedChannelOptions options);
++        public static Channel<T> CreateUnbounded<T>();
++        public static Channel<T> CreateUnbounded<T>(UnboundedChannelOptions options);
++    }
++    public abstract class Channel<T> : Channel<T, T> {
++        protected Channel();
++    }
++    public abstract class Channel<TWrite, TRead> {
++        protected Channel();
++        public ChannelReader<TRead> Reader { get; protected set; }
++        public ChannelWriter<TWrite> Writer { get; protected set; }
++        public static implicit operator ChannelReader<TRead> (Channel<TWrite, TRead> channel);
++        public static implicit operator ChannelWriter<TWrite> (Channel<TWrite, TRead> channel);
++    }
++    public class ChannelClosedException : InvalidOperationException {
++        public ChannelClosedException();
++        public ChannelClosedException(Exception innerException);
++        protected ChannelClosedException(SerializationInfo info, StreamingContext context);
++        public ChannelClosedException(string message);
++        public ChannelClosedException(string message, Exception innerException);
++    }
++    public abstract class ChannelOptions {
++        protected ChannelOptions();
++        public bool AllowSynchronousContinuations { get; set; }
++        public bool SingleReader { get; set; }
++        public bool SingleWriter { get; set; }
++    }
++    public abstract class ChannelReader<T> {
++        protected ChannelReader();
++        public virtual Task Completion { get; }
++        public virtual IAsyncEnumerable<T> ReadAllAsync(CancellationToken cancellationToken = default(CancellationToken));
++        public virtual ValueTask<T> ReadAsync(CancellationToken cancellationToken = default(CancellationToken));
++        public abstract bool TryRead(out T item);
++        public abstract ValueTask<bool> WaitToReadAsync(CancellationToken cancellationToken = default(CancellationToken));
++    }
++    public abstract class ChannelWriter<T> {
++        protected ChannelWriter();
++        public void Complete(Exception error = null);
++        public virtual bool TryComplete(Exception error = null);
++        public abstract bool TryWrite(T item);
++        public abstract ValueTask<bool> WaitToWriteAsync(CancellationToken cancellationToken = default(CancellationToken));
++        public virtual ValueTask WriteAsync(T item, CancellationToken cancellationToken = default(CancellationToken));
++    }
++    public sealed class UnboundedChannelOptions : ChannelOptions {
++        public UnboundedChannelOptions();
++    }
++}
```

Comment:
Nothing here is actually new in Preview7 (most of it isn't new in .NET Core 3.0).  What is new is that the assembly was added in-box, but I'm not sure that means it should show up here as API additions.

Reply:
Thanks for noticing, I think it makes sense to remove the diffs for this then.

Reply:
It might be worth a note somewhere that this is "in-box" now